<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QStackedWidget" name="pages">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1201</width>
      <height>711</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="main_page">
     <widget class="QWidget" name="horizontalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>1191</width>
        <height>702</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="main_page_layout">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QGroupBox" name="groupBox_2">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>50</height>
            </size>
           </property>
           <property name="title">
            <string>цвет фона</string>
           </property>
           <widget class="QWidget" name="horizontalLayoutWidget_5">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>20</y>
              <width>361</width>
              <height>32</height>
             </rect>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QWidget" name="bg_color_preview" native="true">
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>16777215</height>
                </size>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_bg_color_change">
               <property name="text">
                <string>Поменять</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>50</height>
            </size>
           </property>
           <property name="title">
            <string>цвет линии</string>
           </property>
           <widget class="QWidget" name="horizontalLayoutWidget_6">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>20</y>
              <width>361</width>
              <height>32</height>
             </rect>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QWidget" name="line_color_preview" native="true">
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>16777215</height>
                </size>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_line_color_change">
               <property name="text">
                <string>Поменять</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_7">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>50</height>
            </size>
           </property>
           <property name="title">
            <string>Алгоритм</string>
           </property>
           <widget class="QComboBox" name="algorithm_selection">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>20</y>
              <width>379</width>
              <height>32</height>
             </rect>
            </property>
            <item>
             <property name="text">
              <string>ЦДА</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Брезенхем</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Брезенхем целочисленный</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Брезенхем со сглаживанием</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Ву</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Библиотечный</string>
             </property>
            </item>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_3">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>120</height>
            </size>
           </property>
           <property name="title">
            <string>Линия</string>
           </property>
           <widget class="QWidget" name="verticalLayoutWidget_2">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>20</y>
              <width>361</width>
              <height>101</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_4">
               <item>
                <widget class="QLabel" name="label">
                 <property name="minimumSize">
                  <size>
                   <width>90</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Начало (x; y)</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="start_x">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="font">
                  <font>
                   <pointsize>15</pointsize>
                  </font>
                 </property>
                 <property name="text">
                  <string>0</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="start_y">
                 <property name="text">
                  <string>0</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_7">
               <item>
                <widget class="QLabel" name="label_2">
                 <property name="minimumSize">
                  <size>
                   <width>90</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Конец (x; y)</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="end_x">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="font">
                  <font>
                   <pointsize>15</pointsize>
                  </font>
                 </property>
                 <property name="text">
                  <string>0</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="end_y">
                 <property name="text">
                  <string>0</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="btn_line_draw">
               <property name="text">
                <string>Нарисовть</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_6">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>120</height>
            </size>
           </property>
           <property name="title">
            <string>Спектр линий (солнышко)</string>
           </property>
           <widget class="QWidget" name="verticalLayoutWidget">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>20</y>
              <width>361</width>
              <height>101</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>Угол изменения</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="angle">
                 <property name="font">
                  <font>
                   <pointsize>15</pointsize>
                  </font>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_5">
               <item>
                <widget class="QLabel" name="label_4">
                 <property name="text">
                  <string>Радиус</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="radius">
                 <property name="font">
                  <font>
                   <pointsize>15</pointsize>
                  </font>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="btn_spectrum_draw">
               <property name="text">
                <string>Нарисовать</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_4">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="title">
            <string>Управление</string>
           </property>
           <widget class="QWidget" name="verticalLayoutWidget_3">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>20</y>
              <width>361</width>
              <height>291</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <widget class="QLabel" name="lable_scale_center">
               <property name="text">
                <string>Центр масштабирования (0; 0)</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_reset_scale_center">
               <property name="text">
                <string>Сбросить центр масштабирования</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="groupBox_5">
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>70</height>
                </size>
               </property>
               <property name="title">
                <string>Масштаб</string>
               </property>
               <widget class="QWidget" name="horizontalLayoutWidget_2">
                <property name="geometry">
                 <rect>
                  <x>0</x>
                  <y>20</y>
                  <width>351</width>
                  <height>41</height>
                 </rect>
                </property>
                <layout class="QHBoxLayout" name="horizontalLayout_2">
                 <item>
                  <widget class="QSlider" name="scale_slider">
                   <property name="minimum">
                    <number>100</number>
                   </property>
                   <property name="maximum">
                    <number>1500</number>
                   </property>
                   <property name="singleStep">
                    <number>50</number>
                   </property>
                   <property name="pageStep">
                    <number>100</number>
                   </property>
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="tickPosition">
                    <enum>QSlider::TicksAbove</enum>
                   </property>
                   <property name="tickInterval">
                    <number>100</number>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="slider_value">
                   <property name="minimumSize">
                    <size>
                     <width>50</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="font">
                    <font>
                     <pointsize>15</pointsize>
                    </font>
                   </property>
                   <property name="text">
                    <string>100%</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_clear_screen">
               <property name="text">
                <string>Очистить экран</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_9">
               <item>
                <widget class="QLabel" name="label_5">
                 <property name="text">
                  <string>Длина линии</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="line_len"/>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="btn_time_cmp">
               <property name="text">
                <string>Сравнение времени</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_steps_cmp">
               <property name="text">
                <string>Сравнение ступенчатости</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGraphicsView" name="graphicsView">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>800</width>
           <height>700</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>800</width>
           <height>700</height>
          </size>
         </property>
         <property name="backgroundBrush">
          <brush brushstyle="NoBrush">
           <color alpha="255">
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="steps_cmp_page">
     <widget class="QPushButton" name="back_btn">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>100</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Назад</string>
      </property>
     </widget>
     <widget class="QWidget" name="gridLayoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>1181</width>
        <height>661</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="grapchics_layout"/>
     </widget>
    </widget>
    <widget class="QWidget" name="time_cmp_page">
     <widget class="QPushButton" name="back_btn_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>100</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Назад</string>
      </property>
     </widget>
     <widget class="QWidget" name="verticalLayoutWidget_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>1181</width>
        <height>661</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="time_layout"/>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
